cmd/kubeadm/app/util/system/docker_validator.go

```diff
@@ -73,7 +73,7 @@ func (d *DockerValidator) validateDockerInfo(spec *DockerSpec, info types.Info)
 	if !matched {
 		// If it's of the new Docker version scheme but didn't match above, it
 		// must be a newer version than the most recently validated one.
-		ver := `\d{2}\.\d+\.\d+-[a-z]{2}`
+		ver := `\d{2}\.\d+\.\d+(?:-[a-z]{2})?`
```

Comment:
these checks are not super strict.

i think having this is better than the simplified version `\d{2}\.\d+\.\d+`
also, we haven't gotten any reports about the previous regexp not being optimal before.
so this change SGTM.